src/test/scala/uk/gov/hmrc/test/ui/pages/SharedActions.scala

```diff
@@ -17,14 +17,34 @@
 package uk.gov.hmrc.test.ui.pages
 
 import org.openqa.selenium.By
-import uk.gov.hmrc.test.ui.pages.registration.RegistrationLoginPage.authoritySubmitButton
 
 object SharedActions extends BasePage {
 
-  def authorityWizardSubmitButton(): Unit =
-    clickElement(authoritySubmitButton)
+  def clickLinkByPartialText(value: String): Unit ={
+    click(By.partialLinkText(value))
+  }
 
   def assertPartialTextIsDisplayed(value: String): Unit =
     assert(getText(By.id("main-content")).contains(value))
 
+  def clickById(id: String): Unit = {
+    driver.findElement(By.id(id)).click()
+  }
+
+  def selectRadioButtonById(value: String): Unit = {
+    if (value == "") {
+      findElementByCssSelector(".govuk-button").click()
+      assertPartialTextIsDisplayed("Select your UK turnover for {0}")
+  }
+    else if (value == "less-than"){
+      clickById(value)
+      findElementByCssSelector(".govuk-button").click()
+      assertPartialTextIsDisplayed("You do not need to register for the Economic Crime Levy")
+    }
+    else{
+      clickById(value)
+      findElementByCssSelector(".govuk-button").click()
+      assertPartialTextIsDisplayed("Who is your Anti-Money Laundering (AML) supervisor?")
+    }
```

Comment:
this is doing assertions as well as clicking buttons

1. this is a shared action so shouldn't contain lots of branching logic like this.. you should use the shared action from wherever it relates to, that's the reason it's a shared action
2. should this be doing assertions? the assertion should be being made from wherever it's being called

Reply:
Addressed via pairing